### PR TITLE
Support multiple query params

### PR DESCRIPTION
**Pull request by YangJonathan**
No rush to merge. 

Just putting here for awareness.

### CODE REVIEW

File: templates/service.mustache [hunk 1]
```diff
@@ -45,7 +45,7 @@ export function {{FUNCTION_NAME}}(
 ): Promise<AxiosResponse<{{{FUNCTION_RESPONSE}}}>> {
 	return axios.request<{{{FUNCTION_RESPONSE}}}>({
 		headers: headers,
-		url: `${basePathOverride || BASE_PATH}{{{REQUEST_PATH}}}{{#REQUEST_QUERY}}?{{QUERY_NAME}}={{QUERY_VALUE_PATH}}{{/REQUEST_QUERY}}`,
+		url: `${basePathOverride || BASE_PATH}{{{REQUEST_PATH}}}{{#QUERY_FLAG}}?{{/QUERY_FLAG}}{{#REQUEST_QUERY}}{{QUERY_NAME}}={{QUERY_VALUE_PATH}}&{{/REQUEST_QUERY}}`,
 		method: "{{REQUEST_METHOD}}",
```
Comment (YangJonathan) [inline templates/service.mustache:66]: I wasn't sure how this was generated so I manually updated the mustache file. Will need to update w/e it is that need updating to generate this file. 

Also note: to not overcomplicate things this template will leave a trailing `&` after a query param. Upon research, this isn't illegal and is ok. But if we need to change this for whatever reason, I'll update.

File: index.js [hunk 2]
```diff
@@ -110,7 +110,16 @@ exports.generate = async (inputFile, outputDirectory, isApiMonolith, userProvide
 						return parameter.in === "query";
```
Comment (YangJonathan) [inline index.js]: Mmmm... more work needs to be done. I think there will need to be work to support optional query params 


- [x] Investigate obstacles from api service class

File: index.js [hunk 3]
```diff
@@ -106,35 +106,41 @@ exports.generate = async (inputFile, outputDirectory, isApiMonolith, userProvide
 				TYPES_DIRECTORY: isApiMonolith ? "../apiModelTypes" : "./apiModelTypes",
 				BASE_PATH: serviceFileBasePath,
 				FUNCTIONS: _.map(paths, (pathConfig) => {
+					const pathParams = _.filter(pathConfig.parameters, (parameter) => {
+						return _.toLower(parameter.in) === "path";
+					});
```
Comment (YangJonathan) [inline index.js:111]: For some reason I could get incorporate the use of _,filter() directly on line 122 or conditionally returning only queries with parameter.in === "path". So I created this variable to hold them.

**Comment by YangJonathan** [comment]
@MSDOStoevsky  Ready for final review. Can't assign you as a reviewer since I don't have access to the repo

File: templates/service.mustache (hunk 1 above)
Comment (YangJonathan) [reply templates/service.mustache:66]: Comment no longer applicable

File: templates/service.mustache [hunk 4]
```diff
@@ -22,31 +22,49 @@ export interface {{FUNCTION_NAME}}Params {
...
+{{/QUERY_PARAMS}}
+
 /**
  * {{FUNCTION_SUMMARY}}
  * @param params - The request parameters (or undefined).
  * @param payload - the request payload (or undefined).
  * @param headers - the request headers.
  * @param cancelToken - optional cancel token to cancel this request.
  * @param basePathOverride - optional override to the base path.
+ * @param queryParameters - The request query parameters (or undefined).
```
Comment (MSDOStoevsky) [inline templates/service.mustache]: - [ ] Because this parameter is optional, we should mimic the function docs for the other optional params.

Comment (MSDOStoevsky) [reply templates/service.mustache]: - [ ] This argument should be before the basePathOverride

File: index.js [hunk 5]
```diff
@@ -106,35 +106,41 @@ exports.generate = async (inputFile, outputDirectory, isApiMonolith, userProvide
...
 							pathConfig.operationId ||
 							generateOperationId(pathConfig.method, pathConfig.path),
 						FUNCTION_PARAMS: pathConfig.parameters && {
-							FUNCTION_PARAM_CONFIGS: _.map(pathConfig.parameters, (parameter) => {
+							FUNCTION_PARAM_CONFIGS: _.map(pathParams, (parameter) => {
 								return {
 									FUNCTION_PARAM: parameter.name,
-									FUNCTION_PARAM_DESCRIPTION: parameter.description || "stub"
+									FUNCTION_PARAM_DESCRIPTION: parameter.description || "stub",
+									FUNCTION_PARAM_OPTIONAL: parameter.required
```
Comment (MSDOStoevsky) [inline index.js]: - [ ] This value sounds like it will produce the opposite result

File: index.js [hunk 6]
```diff
@@ -106,35 +106,41 @@ exports.generate = async (inputFile, outputDirectory, isApiMonolith, userProvide
...
+								};
+							})
+						},
+						QUERY_PARAMS: queryParams && {
+							QUERY_PARAM_CONFIGS: _.map(queryParams, (parameter) => {
+								return {
+									QUERY_PARAM: parameter.name,
+									QUERY_PARAM_DESCRIPTION: parameter.description || "stub",
+									QUERY_PARAM_OPTIONAL: parameter.required
```
Comment (MSDOStoevsky) [inline index.js]: - [ ] This value sounds like it will produce the opposite result

File: templates/service.mustache [hunk 7]
```diff
@@ -22,31 +22,49 @@ export interface {{FUNCTION_NAME}}Params {
 	/**
 	 * {{FUNCTION_PARAM_DESCRIPTION}}
 	 */
-	{{FUNCTION_PARAM}}: string;
+	{{FUNCTION_PARAM}}{{^FUNCTION_PARAM_OPTIONAL}}?{{/FUNCTION_PARAM_OPTIONAL}}: string;
```
Comment (MSDOStoevsky) [inline templates/service.mustache]: - [ ] Is the swagger documented well enough that we won't see errors throughout with this?

**Review by MSDOStoevsky — CHANGES_REQUESTED** [review]

File: index.js (hunk 5 above)
Comment (YangJonathan) [reply index.js:126]: Fixed.

File: index.js (hunk 6 above)
Comment (YangJonathan) [reply index.js]: Fixed

File: templates/service.mustache (hunk 7 above)
Comment (YangJonathan) [reply templates/service.mustache:25]: Properties not marked as required will default to optional in the generated content.

Comment (YangJonathan) [reply templates/service.mustache]: Fields not marked as required will be set to optional in the generated code. It won't produce an error but can be misleading when implementing a call inside the web app that does require a field but was marked as optional if it was not well-documented in swagger. 

Can't check the checkmark boxes btw lol

File: templates/service.mustache (hunk 4 above)
Comment (YangJonathan) [reply templates/service.mustache]: Fixed

File: index.js (hunk 5 above)
Comment (YangJonathan) [reply index.js]: Fixed

**Review by MSDOStoevsky — APPROVED** [review]
seems good

**Pull request merged**